src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectDpkg.hpp

```diff
@@ -0,0 +1,318 @@
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <limits.h>
+#include <memory>
+#include <regex>
+#include <string>
+
+/**
+ * @brief DpkgVersion data struct.
+ *
+ */
+struct DpkgVersion
+{
+    uint32_t epoch;       ///< Epoch.
+    std::string version;  ///< Version.
+    std::string revision; ///< Revision.
+};
+
+/**
+ * @brief VersionObjectDpkg class.
+ *
+ */
+class VersionObjectDpkg final : public IVersionObject
+{
+private:
+    DpkgVersion struct_dpkg;
+    long m_epoch;
+    std::string m_version;
+    std::string m_revision;
+
+    /**
+     * Give a weight to the character to order in the version comparison.
+     *
+     * @param c An ASCII character.
+     */
+    int order(char c) const
+    {
+        if (std::isdigit(c))
+        {
+            return 0;
+        }
+        else if (std::isalpha(c))
+        {
+            return c;
+        }
+        else if (c == '~')
+        {
+            return -1;
+        }
+        else if (c)
+        {
+            return c + 256;
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    /**
+     * Make the version comparison with the upstream version and the debian revision.
+     *
+     * @param a An array of characters.
+     * @param b An array of characters.
+     */
+    int verrevcmp(const char* a, const char* b) const
+    {
+        if (a == nullptr)
+        {
+            a = "";
+        }
+        if (b == nullptr)
+        {
+            b = "";
+        }
+
+        while (*a || *b)
+        {
+            int first_diff = 0;
+
+            while ((*a && !std::isdigit(*a)) || (*b && !std::isdigit(*b)))
+            {
+                int ac = order(*a);
+                int bc = order(*b);
+
+                if (ac != bc)
+                    return ac - bc;
+
+                a++;
+                b++;
+            }
+            while (*a == '0') a++;
+            while (*b == '0') b++;
+            while (std::isdigit(*a) && std::isdigit(*b))
+            {
+                if (!first_diff)
+                    first_diff = *a - *b;
+                a++;
+                b++;
+            }
+
+            if (std::isdigit(*a))
+                return 1;
+            if (std::isdigit(*b))
+                return -1;
+            if (first_diff)
+                return first_diff;
+        }
+
+        return 0;
+    }
+
+    /**
+     * Compares two Debian versions.
+     *
+     * This function follows the convention of the comparator functions used by
+     * qsort().
+     *
+     * @see deb-version(5)
+     *
+     * @param r_epoch The second epoch.
+     * @param r_version The second version.
+     * @param r_revision The second revision.
+     *
+     * @retval 0 If a and b are equal.
+     * @retval <0 If a is smaller than b.
+     * @retval >0 If a is greater than b.
+     */
+    int dpkg_version_compare(const long r_epoch, const std::string& r_version, const std::string& r_revision) const
```

Comment:
Keep camel case function and arguments naming nomenclature

Reply:
Done!